src/viam/app/data_client.py

```diff
@@ -523,7 +510,7 @@ async def create_dataset(self, name: str, organization_id: str) -> str:
         response: CreateDatasetResponse = await self._dataset_client.CreateDataset(request, metadata=self._metadata)
         return response.id
 
-    async def list_dataset_by_ids(self, ids: List[str]) -> List[Dataset]:
+    async def list_dataset_by_ids(self, ids: List[str]) -> RepeatedCompositeFieldContainer[Dataset]:
```

Comment:
I think we still want to return `List` here and below. We can accomplish that pretty easily by, e.g., `return list(response.datasets)` instead of `return response.datasets)`

Reply:
You could actually return `Sequence` here and no conversion would be necessary 👀